src/Processors/Transforms/Streaming/JoinTransformWithAlignment.cpp

```diff
@@ -289,6 +319,158 @@ void JoinTransformWithAlignment::work()
     }
 }
 
+template <bool is_left_input>
+void JoinTransformWithAlignment::processInputData(LightChunk & chunk)
```

Comment:
I think we shall separate this into 2 functions. Merging them together is super hard to read and there are not a lot of code shared actually.